File: compiler/testData/loadJava/compiledJava/kotlinSignature/propagation/parameter/SubclassFromGenericAndNot.java
package test;

import org.jetbrains.annotations.NotNull;

// Extracted from KT-3302, see Kt3302 test, as well
public interface SubclassFromGenericAndNot {

    public interface NonGeneric {
        void foo(@NotNull String s);

        void dummy(); // to avoid loading as SAM interface
    }

    public interface Generic<T> {
        public void foo(T key);

        void dummy(); // to avoid loading as SAM interface
    }

    public interface Sub extends NonGeneric, Generic<String> {
        @Override
        public void foo(String key);
    }
}

File: kotlin-ide/idea/testData/compiler/loadJava/compiledKotlin/fromLoadJava/kotlinSignature/propagation/parameter/SubclassFromGenericAndNot.kt
package test

// Extracted from KT-3302, see Kt3302 test, as well
public interface SubclassFromGenericAndNot {

    public interface NonGeneric  {
        public fun foo(p0: String)

        public fun dummy() // to avoid loading as SAM interface
    }

    public interface Generic<T>  {
        public fun foo(key: T)

        public fun dummy() // to avoid loading as SAM interface
    }

    public interface Sub : NonGeneric, Generic<String> {
        override fun foo(key: String)
    }
}
